### PR TITLE
Product Info: Thumbnails, Style Name, Size & Quantity Dropdowns, Add to Cart and Favorite Buttons, Styling

### PR DESCRIPTION
**Pull request by WildMountainAir**
Main Files: App, styleThumbnails, styleName, sizeDropdown & quantityDropdown, productInfo, styles.css

Current Bug: I am not passing down the currently selected style through props so my thumbnails when selected all update because there isn't an identifier yet, the style name, and main image carousel have similar issues because of a missing state. This is in my bug log.

### CODE REVIEW

**Review by williams-c — APPROVED** [review]

File: __tests__/__snapshots__/test.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (williams-c) [inline __tests__/__snapshots__/test.spec.js.snap:1]: should gitignore the coverage report

File: __tests__/test.spec.js [hunk 2]
```diff
@@ -1,64 +1,85 @@
...
+  it('renders the Product Details module', () => {
+    expect(wrapper.find(ProductDetails).length).toEqual(1);
   });
 });
 
-describe('TitleImage', () => {
+describe('Title Image', () => {
   test('renders the Title Image', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
```
Comment (williams-c) [inline __tests__/test.spec.js:35]: you can do a before each in your describe blocks for calling mount/shallow and that way to you dont to repeat yourself on lines 39, 43 and 47

File: __tests__/test.spec.js [hunk 3]
```diff
@@ -1,64 +1,85 @@
...
   });
   it('renders the company name', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
     expect(wrapper.find('h1').text()).toEqual('The Proto Company');
   });
   it('renders the informational text', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
     expect(wrapper.find('p').text()).not.toBe('');
```
Comment (williams-c) [inline __tests__/test.spec.js]: is there a reason you aren't checkin that it renders the correct text? not just that it doesn't render the wrong text

**Review by williams-c — APPROVED** [review]

File: __tests__/test.spec.js (hunk 2 above)
Comment (WildMountainAir) [reply __tests__/test.spec.js:35]: That's a great idea!

File: __tests__/test.spec.js (hunk 3 above)
Comment (WildMountainAir) [reply __tests__/test.spec.js]: That's a good point!

**Pull request merged**